ir/src/tests/selectors.rs

```diff
@@ -62,3 +62,104 @@ fn multiconstraint_selectors() {
     let result = AirIR::new(parsed);
     assert!(result.is_ok());
 }
+
+#[test]
+fn selectors_in_evaluators() {
+    let source = "
+    ev evaluator_with_selector([selector, clk]):
+        enf clk' - clk = 0 when selector
+    
+    trace_columns:
+        main: [s[3], clk]
+    
+    public_inputs:
+        stack_inputs: [16]
+    
+    boundary_constraints:
+        enf clk.first = 0
+    
+    integrity_constraints:
+        enf evaluator_with_selector([s[0], clk])";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::new(parsed);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn multiple_selectors_in_evaluators() {
+    let source = "
+    ev evaluator_with_selector([s0, s1, clk]):
+        enf clk' - clk = 0 when s0 & !s1
+    
+    trace_columns:
+        main: [s[3], clk]
+    
+    public_inputs:
+        stack_inputs: [16]
+    
+    boundary_constraints:
+        enf clk.first = 0
+    
+    integrity_constraints:
+        enf evaluator_with_selector([s[0], s[1], clk])";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::new(parsed);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn selector_with_evaluator_call() {
+    let source = "
+    ev unchanged([clk]):
+        enf clk' = clk
+    
+    trace_columns:
+        main: [s[3], clk]
+    
+    public_inputs:
+        stack_inputs: [16]
+    
+    boundary_constraints:
+        enf clk.first = 0
+    
+    integrity_constraints:
+        enf unchanged([clk]) when s[0] & !s[1]";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::new(parsed);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn selectors_inside_match() {
+    let source = "
+    ev is_next_zero([clk]):
+        enf clk' = 0
+
+    ev is_unchanged([clk, s0, s1]):
+        enf clk' = clk when s0 & s1
+
+    ev is_next_one([clk]):
+        enf clk' = 1
+
+    trace_columns:
+        main: [s[3], clk]
+        
+    public_inputs:
+        stack_inputs: [16]
+
+    boundary_constraints:
+        enf clk.first = 0
+
+    integrity_constraints:
+        enf is_next_zero([clk]) when s[0] & !s[1]
+        match enf:
+            is_unchanged([clk, s[0], s[1]])
+            is_next_one([clk]) when !s[0] & !s[1]";
```

Comment:
The point of `match` as I understand it is to only enforce one of these things, depending on the selectors that are specified. Therefore it makes sense to use it when the selectors for different cases are mutually exclusive. In this case, the first evaluator function (which has no selectors) will always be enforced, and the second one never will be, so this doesn't really make sense.